### ISSUE TITLE
Dealing with multi-valued headers in Aleph client

### ISSUE THREAD

**Issue by ddossot** [ztellman/aleph#166]
I'm receiving multiple `Set-Cookie` headers in an HTTP response and I'm trying to retrieve their values as a `seq`.

When I use `(get (:headers response) "set-cookie")`, I receive them as a single coma-separated string (per https://github.com/ztellman/aleph/blob/46ec5c0a9bac1e7bd5ce6fd02033a8736458a8b1/src/aleph/http/core.clj#L124 I believe).

I'd like to avoid having to `,`-split this string to recover the individual values: what approach should I use then?

**Comment by ztellman** [comment]
Previous versions of Aleph provided this as a vector, but the Ring spec actually dictates that they be strings [no matter what](https://github.com/ring-clojure/ring/blob/master/SPEC#L90-L91), so the convention is to make them comma delimited.  Is splitting them apart a performance issue?

**Comment by ddossot** [comment]
I was expecting the [response part of the Ring spec](https://github.com/ring-clojure/ring/blob/master/SPEC#L106-L111) to be at play here.

And no, not concerned about performance issues, more about splitting at the wrong locations since the values are not (somewhat) `,`-escaped before being `,`-joined.

**Comment by ztellman** [comment]
Okay, I think the best thing here is just a separate `get-all` method that behaves the way you expected, but between now and the next release you can just add this to your code:

```clj
(defn get-all [^aleph.http.core.HeaderMap headers ^String k]
  (-> headers ^io.netty.handler.codec.http.HttpHeaders (.headers) (.getAll k)))
```

Leaving this issue open until this is provided in the `aleph.http` namespace.

**Comment by ddossot** [comment]
Thank you sir, this works perfectly :bow:

**Comment by ztellman** [comment]
This is in the master branch, and will be released in `0.4.1`.
